plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/rules/transport/TransportIndexCorrelationRuleAction.java

```diff
@@ -0,0 +1,234 @@
+        private final IndexCorrelationRuleRequest request;
+
+        private final ActionListener<IndexCorrelationRuleResponse> listener;
+
+        AsyncIndexCorrelationRuleAction(IndexCorrelationRuleRequest request, ActionListener<IndexCorrelationRuleResponse> listener) {
+            this.request = request;
+            this.listener = listener;
+        }
+
+        void start() {
+            try {
+                if (!correlationRuleIndices.correlationRuleIndexExists()) {
+                    try {
+                        correlationRuleIndices.initCorrelationRuleIndex(new ActionListener<>() {
+                            @Override
+                            public void onResponse(CreateIndexResponse response) {
+                                try {
+                                    onCreateMappingsResponse(response);
+                                    indexCorrelationRule();
+                                } catch (IOException e) {
+                                    onFailures(e);
+                                }
+                            }
+
+                            @Override
+                            public void onFailure(Exception e) {
+                                onFailures(e);
+                            }
+                        });
+                    } catch (IOException e) {
+                        onFailures(e);
+                    }
+                } else if (!IndexUtils.correlationRuleIndexUpdated) {
+                    IndexUtils.updateIndexMapping(
+                        CorrelationRule.CORRELATION_RULE_INDEX,
+                        CorrelationRuleIndices.correlationRuleIndexMappings(),
+                        clusterService.state(),
+                        client.admin().indices(),
+                        new ActionListener<>() {
+                            @Override
+                            public void onResponse(AcknowledgedResponse response) {
+                                onUpdateMappingsResponse(response);
+                                try {
+                                    indexCorrelationRule();
+                                } catch (IOException e) {
+                                    onFailures(e);
+                                }
+                            }
+
+                            @Override
+                            public void onFailure(Exception e) {
+                                onFailures(e);
+                            }
+                        }
+                    );
+                } else {
+                    indexCorrelationRule();
+                }
+            } catch (IOException ex) {
+                onFailures(ex);
+            }
+        }
+
+        void indexCorrelationRule() throws IOException {
+            IndexRequest indexRequest;
+            if (request.getMethod() == RestRequest.Method.POST) {
+                indexRequest = new IndexRequest(CorrelationRule.CORRELATION_RULE_INDEX).setRefreshPolicy(
```

Comment:
nit: strange formatting. Can we change `setRefreshPolicy` to a newline or does spotless revert back to this oddity?

Reply:
this is due to `spotless`